src/tracker/PatternEditorTools.cpp

```diff
@@ -1718,6 +1718,40 @@ void PatternEditorTools::normalize()
 	
 }
 
+bool PatternEditorTools::selectionContains(const TXMPattern* pattern, const Position& ss, const Position& se, const Position& pos)
+{
+	pp_int32 selectionStartChannel;
+	pp_int32 selectionStartRow;
+	pp_int32 selectionStartInner;
+	pp_int32 selectionEndChannel;
+	pp_int32 selectionEndRow;
+	pp_int32 selectionEndInner;
+	
+	if (!normalizeSelection(pattern, ss, se, 
+							selectionStartChannel, selectionStartRow, selectionStartInner,
+							selectionEndChannel, selectionEndRow,selectionEndInner))
+		return false;
+	
+	// only entire instrument column is allowed
+	if (selectionStartInner >= 1 && selectionStartInner<=2)
```

Comment:
Nitpick: let's try to keep spacing consistent.
`selectionEndInner >= 1 && selectionEndInner<=2`
becomes
`selectionEndInner >= 1 && selectionEndInner <= 2`